File: app/src/main/java/com/haha/binding/BindingAdapters.java
package com.haha.binding;

import android.view.View;
import android.widget.TextView;

import androidx.core.content.ContextCompat;
import androidx.databinding.BindingAdapter;

import kotlin.Pair;

public final class BindingAdapters {

    private BindingAdapters() {
    }

    @BindingAdapter("visibleGone")
    public static void setVisibleGone(View view, Boolean visible) {
        view.setVisibility(Boolean.TRUE.equals(visible) ? View.VISIBLE : View.GONE);
    }

    @BindingAdapter("labelText")
    public static void setLabelText(TextView textView, Pair<String, String> pair) {
        if (pair == null) {
            textView.setText("");
            return;
        }
        textView.setText(pair.getFirst() + "：" + pair.getSecond());
    }

    @BindingAdapter("tagColorRes")
    public static void setTagColor(TextView textView, int colorRes) {
        if (colorRes == 0) {
            return;
        }
        int color = ContextCompat.getColor(textView.getContext(), colorRes);
        textView.setBackgroundColor(color);
    }
}

File: app/src/main/java/com/haha/binding/BindingAdapters.kt
package com.haha.binding

import android.view.View
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.databinding.BindingAdapter

object BindingAdapters {

    @JvmStatic
    @BindingAdapter("visibleGone")
    fun setVisibleGone(view: View, visible: Boolean?) {
        view.visibility = if (visible == true) View.VISIBLE else View.GONE
    }

    @JvmStatic
    @BindingAdapter("labelText")
    fun setLabelText(textView: TextView, pair: Pair<String, String>?) {
        if (pair == null) {
            textView.text = ""
            return
        }
        textView.text = "${pair.first}：${pair.second}"
    }

    @JvmStatic
    @BindingAdapter("tagColorRes")
    fun setTagColor(textView: TextView, colorRes: Int?) {
        if (colorRes == null || colorRes == 0) return
        val color = ContextCompat.getColor(textView.context, colorRes)
        textView.setBackgroundColor(color)
    }
}
